src/late/npm-on-ipfs/scripts/npm.js

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env node
+
+const homedir = require('os').homedir()
+const { readFile } = require('fs')
+const { execFile } = require('child_process')
+const { join } = require('path')
+
+readFile(join(homedir, '.ipfs/api'), (err, data) => {
+  const pkg = process.argv[1].includes('npm') ? 'npm' : 'yarn'
+  const addr = data.toString()
+  let args = [
+    ...process.argv.slice(2, process.argv.length),
+    '--package-manager', join(__dirname, `./${pkg}.bak`)
```

Comment:
@olizilla @lidel I need your advise here. We need to tell our `npm-on-ipfs` binary where is the actual package manager to use. The issue is: where should we back it up?

For the IPFS CLI tools, we back them up to IPFS Desktop user data directory, which won't be removed during any installations/uninstallation, which is good. And we can pass that directory path along to the scripts.

Although, in this case, `npm.js` will be executed plainly as `npm` or `yarn` so we can't tell where the user directory is. One possibility I see is to add to `/usr/local/bin` a file with the hardcoded path to the yarn and npm backups and we could keep this file as template.

Right now I'm just throwing it in the source directory, which is neither a good practice, nor useful since it would be removed by deleting the app. What do you think? **Where should we back npm/yarn to?** And how to handle that?

Reply:
Is the underlying problem "I don't know the path to `npm`" ?

On linux/mac there are two handy commands:
- `where npm` shows _all_ `npm` binaries across directories listed in `PATH` 
- `which npm` shows which binary is selected when you execute `npm`

I believe `where` is also on Windows.

(If I misunderstood your underlying problem, I'll need an example to illustrate it)

Reply:
I think the question is: 

_if we move the original `npm` or  `yarn`  commands off the `PATH`, to replace them with the `ipfs-npm` proxy, where should we stash them so we can revert the change later?_

Reply:
The question is a mix of what @olizilla said and something else: where should we stash them so we can tell ipfs-npm to use them? Because we need to pass the original binary through `package-manager`, otherwise it will end up in a loop.

Reply:
gotcha... I'd do something like move it to `_npm`. it feels gross though. Worth some further thinking... like what are the other options

1. dont mess with the users PATH... add `ipfs-npm` but swap out `npm`
2. move `npm` to `_npm` so we can still call  it from `ipfs-npm`
3. bundle a version of `npm`  with `ipfs-npm` so it can reliably call that one.

The more I think about the worse option 2 sounds. I would go with option 1 for now, which simplifies this PR and makes it less risky. Also, we still need to do work on `ipfs-npm` to  make it more user friendly, so it make sense to **not** swap it out for the regular `npm` yet... it's still experimental, so it's not ready for prime time yet.

Once it is more stable, option 3 might be a better way to go, so that ipfs-npm is proxying a known version of `npm` rather than whatever the user happens to have installed.

Reply:
So you're suggesting just to install npm-ipfs package right?

Reply:
My 2 cents: remember to do a quick sanity check when deciding how to override `npm.

Namely, is **moving binaries belonging to other packages installed in the system** is better than **adding a directory in front of `PATH` variable**?
Think about scenario when someone uninstalls `npm`. Our proxy be removed. 

Unless I am, missing something, 
adding a directory to the front of `PATH` seems the only "safe" way to override.